posts/inside-rust/2024-06-19-this-development-cycle-in-cargo-1.80.md

```diff
@@ -0,0 +1,517 @@
+([#14016](https://github.com/rust-lang/cargo/issues/14016))
+.
+We considered whether we should deprecate the syntax (which we had previously deferred) or whether to implicitly inject `"dep:dep_name"`.
+After some discussion, we settled on the latter.
+
+##### Normalizing published manifest files
+
+*Update from [1.79](https://blog.rust-lang.org/inside-rust/2024/05/07/this-development-cycle-in-cargo-1.79.html#normalizing-published-package-files)*
+
+Now that published `Cargo.toml` files list all targets,
+we can avoid reading the filesystem with [#13849](https://github.com/rust-lang/cargo/pull/13849).
+
+Thanks for [stormshield-guillaumed](https://github.com/stormshield-guillaumed) testing nightlies,
+we found out that these changes made the vendored `Cargo.toml` non-deterministic and fixed that in [#14004](https://github.com/rust-lang/cargo/pull/14004).
+
+##### Merging `cargo upgrade` into `cargo update`
+
+*Update from [1.77](https://blog.rust-lang.org/inside-rust/2024/02/13/this-development-cycle-in-cargo-1-77.html#merging-cargo-upgrade-into-cargo-update)*
+
+In [#13979](https://github.com/rust-lang/cargo/pull/13979), [torhovland](https://github.com/torhovland) added support for `cargo update --breaking` with the following policy:
+1. In-memory, upgrades certain workspace member dependencies to the latest breaking version
+  - Must use `^` version requirement operator (the default)
+  - Must not be renamed as those tend to be used to allow multiple versions of the same package and the user likely doesn't want the version changed
+  - Can further limit to specific dependencies by naming them
+2. Reconcole the lockfile with these new version requirements
+3. Write out the new version requirements, preserving the precision used in the original requirement (e.g. `1.0` would be upgraded to `2.0`, not `2.0.5`)
+
+This is available to use as of nightly TODO
+
+##### `.crate` provenance
+
+With the xz backdoor, there has been an increased interest in verifying that a published `.crate` matches the repo.
+Cargo already includes a `cargo_vcs_info.json` file in the `.crate` to identify what `cargo publish` was run against.
+One problem is that this file wasn't being generated if `--allow-dirty` was used
+([#13695](https://github.com/rust-lang/cargo/issues/13695)).
+
+After some discussion, [torhovland](https://github.com/torhovland) created [#13960](https://github.com/rust-lang/cargo/pull/13960) for us to always generate `cargo_vcs_info.json` but to include a `dirty: "true"` field when `--allow-dirty` was used.
+
+For more investigation on using this file, see
+[999 crates of Rust on the wall](https://lawngno.me/blog/2024/06/10/divine-provenance.html).
+
+##### `cargo publish --workspace`
+
+[torhovland](https://github.com/torhovland) stepped up to break down what it would take to add `--workspace`  in [#1169](https://github.com/rust-lang/cargo/issues/1169) and [#10948](https://github.com/rust-lang/cargo/issues/10948) and
```

Comment:
@jneem and I collaborated on the first part, including the initial #13947. But since the overlay solution came up, it's been all @jneem.

Reply:
Not sure the exact wording that would be accurate here but I made an attempt

Reply:
It could mention @jneem and me already on line 272. Then it's fine.

Reply:
How is b80fefa ?